openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1119,6 +1121,92 @@ Cache-Control: no-store
 }
 ```
 
+# Callback Endpoint {#callback_endpoint}
+
+This endpoint is used by the Wallet to notify the Credential Issuer whether a credential has been successfully received or not. The Credential Issuer needs to return `callback_id` in the Credential Response or a Batch Credential Response for the Wallet to be able to use this Endpoint. Support for this endpoint is OPTIONAL.
+
+This endpoint can be used after the Credential Issuer has sent Credential Response or Batch Credential Response. It enables the Credential Issuer to take subsequent actions after issuance, depending on whether the Credential has been accepted and successully stored by the Wallet, rejected by the Wallet, or errors and other unforeseen circumstances have occurred during the Wallet's processing.
+
+The Wallet MUST present to the Callback Endpoint a valid Access Token issued at the Token Endpoint as defined in (#token_endpoint). Credential Issuer that requires request to the Callback Endpoint MUST ensure Access Token issued by the Authorization Server is valid at the Callback Endpoint.
+
+The callback from the Wallet is idempotent. The Credential Issuer MUST return success if it receives multiple identical calls from the Wallet for the same `callback_id`s.
+
+It is up to the Wallet whether to retry if the call to this endpoint fails. The Credential Issuer SHOULD have pre-determined the amount of time within with it expects the callback. It is up to the Credential Issuer how to interpret not receiving callback from the Wallet at all, despite providing `callback_id` in the Credential Response.
+
+Communication with the Callback Endpoint MUST utilize TLS.
+
+## Callback from the Wallet {#callback}
+
+The Wallet sends an HTTP POST request to the Callback Endpoint with the following parameters in the entity-body and using the `application/json` media type.
+
+* `credentials`: A JSON array of objects, where each object consists of the following parameters:
+  * `callback_id`: REQUIRED. A JSON string received in Credential Response or Batch Credential Response.
+  * `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `success`, `failure` or `rejected`. `rejected` is be used when unsuccessful credential issuance was caused by user action. In all other cases, `failure` is used.
+  * `error_description`: OPTIONAL. Human-readable ASCII [@!USASCII] text providing additional information, used to assist the Credential Issuer developer in understanding the error that occurred. Values for the `error_description`` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+Below is a non-normative example of a callback request when credential issuance was successful:
+
+```
+POST /callback HTTP/1.1
+Host: server.example.com
```

Comment:
Both examples should probably have `Authorization: Bearer mF_9.B5f-4.1JqM` added.

Reply:
done